### PR TITLE
Test utility to assert that a single UI action only generates a single undo step

### PR DESCRIPTION
**Pull request by bkrmendy**
## Description
This PR adds a test utility which enables us to assert that a given UI action only generates a single undo.

This implemented by adding a helper, `expectSingleUndoStep`, which checks the size of `EditorStorePatched.history.previous` before and after a given UI action, and expects the size difference to be 1 (meaning that 1 new undo frame was added).

### CODE REVIEW

File: editor/src/components/canvas/event-helpers.test-utils.tsx [hunk 1]
```diff
@@ -694,3 +695,13 @@ export function switchDragAndDropElementTargets(
...
+
+export async function expectSingleUndoStep<T>(
+  editor: EditorRenderResult,
+  action: () => Promise<T>,
+): Promise<T> {
+  const undoCount = editor.getUndoCount()
+  const result = await action()
+  expect(editor.getUndoCount()).toEqual(undoCount + 1)
+  return result
+}
```
Comment (Rheeseyb) [inline editor/src/components/canvas/event-helpers.test-utils.tsx:707]: The function itself is fine, but this file is the wrong place for it. I _think_ `ui-jsx.test-utils.tsx` would be better 🤔 (unless you can find a better place)

Comment (bkrmendy) [reply editor/src/components/canvas/event-helpers.test-utils.tsx]: I moved it to `utils.test-utils.ts` in [e461839](https://github.com/concrete-utopia/utopia/pull/3178/commits/e461839dff9fa75dd25902d49ef5e623b60e7f1d)

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/canvas/ui-jsx.test-utils.tsx [hunk 2]
```diff
@@ -199,6 +200,7 @@ export async function renderTestEditorWithModel(
   loginState: LoginState = notLoggedIn,
 ): Promise<EditorRenderResult> {
   const renderCountBaseline = renderCount
+  let undoCount = 0
```
Comment (balazsbajorics) [inline editor/src/components/canvas/ui-jsx.test-utils.tsx:203]: seems to be leftover

File: editor/src/components/canvas/event-helpers.test-utils.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { emptyModifiers, Modifiers } from '../../utils/modifiers'
 import { resetMouseStatus } from '../mouse-move'
 import keycode from 'keycode'
 import { assertNever, NO_OP } from '../../core/shared/utils'
+import { EditorRenderResult } from './ui-jsx.test-utils'
```
Comment (balazsbajorics) [inline editor/src/components/canvas/event-helpers.test-utils.tsx:6]: leftover

**Review by balazsbajorics — APPROVED** [review]
very nice! easy to use API and easy to read code

**Review by gbalint — APPROVED** [review]

**Review by enidemi — APPROVED** [review]

**Pull request merged**